packages/features/data-table/components/filters/DateRangeFilter.tsx

```diff
@@ -154,38 +162,58 @@ export const DateRangeFilter = ({ className, column }: DateRangeFilterProps) =>
-        disabled={false}
-        onDatesChange={updateDateRangeFromPicker}
-      />
-      <Select
-        variant="default"
-        data-testid="insights-preset"
-        options={PRESET_OPTIONS.map((o) => ({
-          label: t(o.labelKey, o.i18nOptions),
-          value: o.value,
-        }))}
-        value={{ label: t(selectedPreset.labelKey, selectedPreset.i18nOptions), value: selectedPreset.value }}
-        className="w-40 capitalize text-black"
-        defaultValue={{
-          label: t(selectedPreset.labelKey, selectedPreset.i18nOptions),
-          value: selectedPreset.value,
-        }}
-        styles={{ menuPortal: (base) => ({ ...base, zIndex: 9999 }) }}
-        menuPortalTarget={document.body}
-        onChange={(e) => {
-          if (e) {
-            updateDateRangeFromPreset(e.value);
-          }
-        }}
-      />
-    </div>
+    <Popover>
+      <PopoverTrigger asChild>
+        <Button color="secondary" className="items-center capitalize">
+          {selectedPreset.value !== CUSTOM_PRESET_VALUE && (
+            <span>{t(selectedPreset.labelKey, selectedPreset.i18nOptions)}</span>
+          )}
+          {selectedPreset.value === CUSTOM_PRESET_VALUE &&
+            (endDate ? (
+              <span>
+                {format(startDate.toDate(), "LLL dd, y")} - {format(endDate.toDate(), "LLL dd, y")}
+              </span>
+            ) : (
+              <span>{format(startDate.toDate(), "LLL dd, y")} - End</span>
+            ))}
+          <Icon name="chevron-down" className="ml-2 h-4 w-4" />
+        </Button>
+      </PopoverTrigger>
+      <PopoverContent className="flex w-fit p-0" align="end">
+        {selectedPreset.value === CUSTOM_PRESET_VALUE && (
+          <div className="border-subtle border-r">
+            <DateRangePicker
+              dates={{
+                startDate: startDate.toDate(),
+                endDate: endDate?.toDate(),
+              }}
+              minDate={currentDate.subtract(2, "year").toDate()}
+              maxDate={currentDate.toDate()}
+              disabled={false}
+              onDatesChange={updateDateRangeFromPicker}
+              withoutPopover={true}
+            />
+          </div>
+        )}
+        <Command className="w-40">
+          <CommandList>
+            {PRESET_OPTIONS.map((option) => (
+              <CommandItem
+                key={option.value}
+                className={classNames(
+                  "cursor-pointer justify-between",
+                  selectedPreset.value === option.value && "bg-emphasis"
+                )}
+                onSelect={() => {
+                  updateDateRangeFromPreset(option.value);
+                }}>
+                <span className="capitalize">{t(option.labelKey, option.i18nOptions)}</span>
+                {selectedPreset.value === option.value && <Icon name="check" />}
+              </CommandItem>
+            ))}
+          </CommandList>
+        </Command>
```

Comment:
NIT: Can we double check the styling of this section - seems like we should have a bit of padding and gap between elements <3